packages/web3-eth-ens/src/ENS.js

```diff
@@ -0,0 +1,162 @@
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    web3.js is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+    You should have received a copy of the GNU Lesser General Public License
+    along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @file ENS.js
+ *
+ * @author Samuel Furter <samuel@ethereum.org>
+ * @date 2018
+ */
+
+"use strict";
+
+var config = require('./config');
+var Registry = require('./contracts/Registry');
+
+/**
+ * varructs a new instance of ENS
+ *
+ * @method ENS
+ * @param {Object} eth
+ * @varructor
+ */
+function ENS(eth) {
+    this.eth = eth;
+}
+
+Object.defineProperty(ENS.prototype, 'registry', {
+    get: function(){
+        return new Registry(this);
+    },
+    enumerable: true
+});
+
+/**
+ * Returns the address record associated with a name.
+ *
+ * @method getAddress
+ * @param {string} name
+ * @return {Promise<string>} a promise
+ */
+ENS.prototype.getAddress = function (name) {
+    return this.registry.resolver(name).then(function(resolver) {
+        return resolver.addr();
+    }).catch(function(error) {
+        throw error;
+    });
+};
+
+/**
+ * Sets a new address
+ *
+ * @param {string} name
+ * @param {string} address
+ * @param {string} from
+ * @returns {Promise<Transaction>}
+ */
+ENS.prototype.setAddress = function (name, address, from) {
+    return this.registry.resolver(name).then(function(resolver) {
+        return resolver.setAddr(address, from);
+    }).catch(function(error) {
+        throw error;
+    });
+};
+
+/**
+ * Returns the public key
+ *
+ * @param {string} name
+ * @returns {Promise<T>}
+ */
+ENS.prototype.getPubkey = function(name) {
+  return this.registry.resolver(name).then(function(resolver) {
+      return resolver.pubkey();
+  }).catch(function(error) {
+      throw error;
+  });
+};
+
+/**
+ * Set the new public key
+ *
+ * @param {string} name
+ * @param {string} x
+ * @param {string} y
+ * @param {string} from
+ * @returns {Promise<Transaction>}
+ */
+ENS.prototype.setPubkey = function(name, x, y, from) {
+  return this.registry.resolver(name).then(function(resolver) {
+      return resolver.setPubkey(x, y, from);
+  }).catch(function(error) {
+      throw error;
+  });
+};
+
+/**
+ * Returns the content
+ *
+ * @param {string} name
+ * @returns {Promise<T>}
+ */
+ENS.prototype.getContent = function(name) {
+    return this.registry.resolver(name).then(function(resolver) {
+        return resolver.content();
+    }).catch(function(error) {
+        throw error;
+    });
+};
+
+/**
+ * Sets the new content
+ *
+ * @param {string} name
+ * @param {string} hash
+ * @param {string} from
+ * @returns {Promise<Transaction>}
+ */
+ENS.prototype.setContent = function(name, hash, from) {
```

Comment:
Could you add support for multihash/setMultihash too, please?

Reply:
What are the respective function names? Its not in the current ABI

Reply:
https://eips.ethereum.org/EIPS/eip-1062